lib/ocpp/v201/smart_charging.cpp

```diff
@@ -102,8 +121,34 @@ ProfileValidationResultEnum SmartChargingHandler::validate_profile_schedules(con
     return ProfileValidationResultEnum::Valid;
 }
 
-void SmartChargingHandler::add_profile(const ChargingProfile& profile) {
-    charging_profiles.push_back(profile);
+void SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
+    if (STATION_WIDE_ID == evse_id) {
+        station_wide_charging_profiles.push_back(profile);
+    } else {
+        charging_profiles.push_back(EvseProfile{evse_id = evse_id, profile = profile});
+    }
+}
+
+std::vector<ChargingProfile> SmartChargingHandler::get_evse_specific_tx_default_profiles() const {
```

Comment:
If there are evse specific default profiles, wouldn't it make sense to filter on them as well? Now you return all profiles for all evse's. I would expect a test to expose this issue as well.

Reply:
I'm unclear on what issue you are referring to. Are you saying that it shouldn't return all profiles for all EVSEs?

Otherwise, while you might be right that it might make sense to eventually filter based on EVSE, it wasn't necessary for this code, and I did not want to write code for something I might anticipate a need for, increasing the size of the PR and distracting from the focus of it.

Unless I misunderstand what you mean?

Reply:
It could very well be that I misunderstood the intended functionality here.

What I meant is that if you add a TxDefaultProfile for EVSE 1 and then try to add a TxDefaultProfile for EVSE 2 I would expect that to be possible. Interpreting the code here I expect you to get a `ProfileValidationResultEnum::DuplicateTxDefaultProfileFound` error instead.

Reply:
Ah, I see what you are saying now. You're right. I can add a test case for that case.

Reply:
Turned out that it was already going to pass.

Reply:
I guess it now passes since you are adding another profile with a different `ChargingProfile.id`. This makes sense now I've read this part of the spec because we can never have multiple profiles with the same ID according to usecase K01.FR.05. I did not know that before.